File: Player/PlayerAudio.js
// This script manage all the audio of the player.
var mainAS : AudioSource;
//var runAS : AudioSource;

private var clip : AudioClip;

var jumpSound : AudioClip;
var hurtSound : AudioClip;
var deathSound : AudioClip;

var isPlayed : boolean;

function Update() {
    var isGrounded = GetComponent(PlayerControl).isGrounded;
    var playerCanMove = GetComponent(PlayerControl).playerCanMove;
    var isDead = GetComponent(PlayerHealth).isDead;

    if (playerCanMove) {
        if (Input.GetButtonDown("Jump") && isGrounded) {
            mainAS.PlayOneShot(jumpSound);
        }
        /*
        // Running sound work but the sound overlap, only one sound with variation should be played.
        if (Input.GetAxis("Horizontal") != 0 && isGrounded) {
            if (!runAS.isPlaying) {
                runAS.Play();
            }
        }*/
    }

    if (isDead && !isPlayed) {
        Camera.main.GetComponent(AudioSource).enabled = false;
        clip = deathSound;
        PlaySound(clip);
    }
}

function OnTriggerEnter2D (other : Collider2D) {
    if (!other.isTrigger) {
        if (other.gameObject.tag == "Enemy") {
            if(!GetComponent(PlayerHealth).isDead && GetComponent(PlayerHealth).playerHealth > 0.3) {
                mainAS.PlayOneShot(hurtSound);
            }
        }
    }
}

function PlaySound(clip : AudioClip) {
    mainAS.PlayOneShot(clip);
    isPlayed = true;
    yield WaitForSeconds(2); // When the clip played fully change the isPlayed state.
    isPlayed = false;
}

File: Player/PlayerAudio.cs
// using System.Collections;
// using System.Collections.Generic;
// using UnityEngine;

// public class PlayerAudio : MonoBehaviour {
// 	AudioSource audiosource;
// 	public AudioClip air;
// 	public AudioClip attack;
// 	public AudioClip hurt;
// 	public AudioClip die;
// 	public bool wasPlayed = false;
// 	PlayerHealth playerHealth;


// 	void Start () {
// 		audiosource = GetComponent<AudioSource>();
// 		playerHealth = GetComponent<PlayerHealth>();
// 	}
	
// 	void Update () {
// 		StartCoroutine(Soundify());
// 	}

// 	void Soundify() {
// 		// SOUNDS SHOULD BE PLAYED ACCORDING TO THE PLAYER ANIMATIONS INSTEAD OF BUTTON PRESS?
// 		// Recheck this script	
// 		if(!playerHealth.isDead && GetComponent<PlayerControl>().canMove) {
// 			if(Input.GetButtonDown("Jump") && GetComponent<PlayerControl>().isGrounded) {
// 				audiosource.PlayOneShot(air);
// 			}

// 			if(Input.GetButtonDown("Attack") && !GetComponent<PlayerAttack>().isAttacking) {
// 				audiosource.PlayOneShot(attack);
// 			}
// 		}

// 		// if(playerHealth.isHurt && !wasPlayed) {
// 		// 	wasPlayed = true;
// 		// 	audiosource.PlayOneShot(hurt);
// 		// 	yield return new WaitForSeconds(1);
// 		// 	wasPlayed = false;
// 		// }
		
// 		if(GetComponent<Animator>().GetBool("die") && !wasPlayed) {
// 			wasPlayed = true;
// 			audiosource.PlayOneShot(die);
// 		}
// 	}
// }
